src/events/cd-event-add-child-ticket.vue

```diff
@@ -0,0 +1,182 @@
+          <option value="prefer not to answer">{{ $t('Prefer not to answer') }}</option>
+          <option value="specify">{{ $t('Specify Identity') }}</option>
+        </select>
+        <input v-show="specifyGender" class="form-control" v-model="genderInput" :placeholder="$t('Identify as...')" data-vv-name="gender" data-vv-validate-on="blur" v-validate="specifyGender ? 'required' : ''"/>
+      </div>
+      <p class="cd-child-ticket__gender-err text-danger" v-show="errors.has('gender:required')">{{ $t('Gender is required') }} <a v-on:click="showWhy">{{ $t('Why is this required? Click Here To Find Out More') }}</a><p v-show="whyGender && errors.has('gender:required')">{{ $t(`We want to provide activities that appeal to people regardless of their gender. To check how well we are succeeding, we'd like to find out whether or not people of different genders are equally likely to take part.`) }}</p></p> 
+
+      <label>{{ $t('Ticket') }}</label>
+      <div class="cd-child-ticket__ticket-selector"> 
+        <multiselect v-model="tickets" :options="childTickets" group-label="name" group-values="tickets" :multiple="true" :searchable="false" :group-select="false" :placeholder="$t('Select Event Tickets')" track-by="id" label="name" @close="onTicketTouch" ></multiselect>
+      </div>
+      <p class="cd-child-ticket__ticket-select-err text-danger" v-show="invalidTicket">{{ $t('Ticket selection is required') }}</p>  
+    </div>
+  </div>
+</template>
+
+<script>
+  import VueDobPicker from 'vue-dob-picker';
+  import Multiselect from 'vue-multiselect';
+
+  export default {
+    name: 'ChildTicket',
+    props: ['sessions', 'eventId', 'event'],
+    components: {
+      VueDobPicker,
+      Multiselect,
+    },
+    data() {
+      return {
+        firstName: '',
+        surame: '',
+        dob: null,
+        genderSelect: '',
+        genderInput: '',
+        whyGender: false,
+        ticketTouch: false,
+        tickets: '',
+      };
+    },
+    methods: {
+      showWhy() {
+        this.whyGender = true;
+      },
+      onTicketTouch() {
+        this.ticketTouch = true;
+      },
+    },
+    filters: {
+      addPosession(firstName) {
+        if (firstName) {
+          return ` - ${firstName}`;
+        }
+        return firstName;
+      },
+    },
+    computed: {
+      childTickets() {
+        return this.sessions.map(session => ({
+          description: session.description,
+          entity$: session.entity$,
+          eventId: session.eventId,
+          id: session.id,
+          name: session.name,
+          status: session.status,
+          tickets: session.tickets.filter(ticket => ticket.type === 'ninja'),
+        }));
+      },
+      name() {
+        return `${this.firstName} ${this.surame}`;
+      },
+      specifyGender() {
+        return this.genderSelect === 'specify';
+      },
+      gender() {
+        this.whyGender = false;
+        if (this.genderSelect === 'specify') {
```

Comment:
You could use the same v-model for both scenario. In the end, i believe it'll just store a string

Reply:
Tried that! Came across a scenario where if you blurred out of the input field for specifying gender, it would give the required error, and then if you selected another gender (ex male) the error would still show because that required error is still there for the input field even though it wasn't being shown any longer. So this is what was used instead

Reply:
I tried that as well. The issue for the if/show is that the gender has to be equal to the value 'specify' and when the input box shows it already has 'specify' in it as the gender. once i try to change it to anything else the box won't show (as the gender will not be equal to 'specify'). so genderSelect and genderInput are used and then gender is computed from there

Reply:
I guess onTouch you could reset the value of gender from specify to null ?

Reply:
But the input field shows based on the condition that the value of gender is specify, so if it is reset then it wouldn't show at all